pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -516,11 +611,30 @@ func (p *PriorityQueue) AddUnschedulableIfNotPresent(pInfo *framework.QueuedPodI
 	for plugin := range pInfo.UnschedulablePlugins {
 		metrics.UnschedulableReason(plugin, pInfo.Pod.Spec.SchedulerName).Inc()
 	}
-	if p.moveRequestCycle >= podSchedulingCycle {
+	// While the pod was in flight, a more recent cycle number might have
+	// been recorded for it in p.inFlightPods, so in that case we need to
+	// honor the more recent cycle number.
+	moveRequestCycle := pInfo.MoveRequestCycle
+	inFlightMoveRequestCycle := p.inFlightPods[pInfo.Pod.UID].moveRequestCycle
+	if inFlightMoveRequestCycle > moveRequestCycle {
+		moveRequestCycle = inFlightMoveRequestCycle
+	}
+
+	// inFlightPod.successRequeuePlugins has plugins that may change its scheduling result by events that happened during scheduling.
+	// So, here checks whether any of UnschedulablePlugins are in inFlightPod.successRequeuePlugins,
+	// and if so, enqueue this Pod to backoffQ.
+	if moveRequestCycle >= podSchedulingCycle && p.inFlightPods[pInfo.Pod.UID].successRequeuePlugins.HasAny(pInfo.UnschedulablePlugins.UnsortedList()...) {
```

Comment:
Here, I forgot to think about the cases:
- unschedulablePlugins aren't supported Requeue.
- unschedulablePlugins is empty.

I'll update tomorrow.

Reply:
@pohly 

> So the pod was listed in p.unschedulablePods, but its pInfo.UnschedulablePlugins was empty. Is that a valid combination?

I guess that could happen due to ^ my mistake. It's fixed now.

Reply:
With the revised code, if all plugins returned "okay to requeue" but then `pInfo.UnschedulablePlugins` is empty, the pod will go to the backoff queue, right?

That then doesn't address the problem that I raised because there are still false positives.

Reply:
Yes, right. What's the expected behavior in empty unschedulable plugins case then?
If the Pod has no unschedulable plugin, the scheduling cycle was failed with unusual error. (= not FitError)
I think we should follow the current way of requeuing for such Pods because events won't help to make them schedulable.

Reply:
> What's the expected behavior in empty unschedulable plugins case then?

The more precise filtering at the cluster event level (= https://github.com/kubernetes/kubernetes/pull/115796) detects that a concurrent cluster event is irrelevant and then doesn't set `moveRequestCycle` in the first place. Then this code here can put the pod into the unscheduable queue, just as it would if the event hadn't occurred.

> I think we should follow the current way of requeuing for such Pods because we cannot know which event will help to make them schedulable.

We don't need to know which events make the pod schedulable - it's sufficient to know which events *don't* make them schedulable. If some event occurs that triggers scheduling, great! If not, periodic flushing with `flushUnschedulablePodsLeftover` will trigger new attempts.

Reply:
Sorry. I don't understand yet.

>  it's sufficient to know which events don't make them schedulable. 

Which plugin gives such a suggestion? In your case, SchedulingHint from which plugin decides this event doesn't make Pod schedulable?
If a plugin is part of the cause of scheduling failure of that Pod, then that plugin should be registered as unschedulable plugins. 
And, "which event make such Pods schedulable" or "which event doesn't make them schedulable"; they can only be decided by plugins registered in unschedulable plugins. This is my point. Do I miss something?

Reply:
> Which plugin gives such a suggestion?

With hints, a plugin can say "I don't care about this event". If all plugins agree, the event can be ignored.

Ruling out events is often easy and cheap. An exact response for "I am waiting for this event" is harder because it depends on knowing exactly what is blocking scheduling.

> If a plugin is part of the cause of scheduling failure of that Pod, then that plugin should be registered as unschedulable plugins.

I don't think that all scheduling failures cause the plugin to be added to `pInfo.UnschedulablePlugins`. I only see one place where something gets added to `pInfo.UnschedulablePlugins`, which is in `runPreEnqueuePlugins`. When the DRA plugin fails, it's usually during `Reserve`.